src/components/scoring/scoring.service.js

```diff
@@ -0,0 +1,126 @@
+        }
+
+        function _isTypeFull(typeId) {
+            return ExamTypeEnum['FULL TEST'].enum === typeId;
+        }
+
+        function _getScoreTableKeyByTypeId(typeId) {
+            return _isTypeFull(typeId) ? keysMapConst.test : keysMapConst.miniTest;
+        }
+
+        function _getDataFromTable(scoreTable, key, id, rawScore) {
+            var data = angular.copy(scoreTable);
+            if (angular.isDefined(key)) {
+                data = data[key];
+            }
+            if (angular.isDefined(id)) {
+                data = data[id];
+            }
+            if (angular.isDefined(rawScore)) {
+                data = data[rawScore];
+            }
+            return data;
+        }
+
+        function _getResultsFn(scoreTable, questionsResults, typeId, id) {
+            var rawScore = _getRawScore(questionsResults);
+            var key = _getScoreTableKeyByTypeId(typeId);
+            return _getDataFromTable(scoreTable, key, id, rawScore);
+        }
+
+        function _getTestScoreResultFn(scoreTable, questionsResults, typeId, categoryId) {
+            var data = _getResultsFn(scoreTable, questionsResults, typeId, categoryId);
+            return {
+                testScore: data
+            };
+        }
+
+        function _getSectionScoreResultFn(scoreTable, questionsResults, typeId, subjectId) {
+            var data = _getResultsFn(scoreTable, questionsResults, typeId, subjectId);
+            return {
+                sectionScore: data
+            };
+        }
+
+        // api
+
+        scoringServiceObjApi.isTypeFull = function (typeId) {
+            return ExamTypeEnum['FULL TEST'].enum === typeId;
+        };
+
+        scoringServiceObjApi.getTestScoreResult = function (questionsResults, typeId, categoryId) {
+            return _getScoreTableProm().then(function (scoreTable) {
+                return _getTestScoreResultFn(scoreTable, questionsResults, typeId, categoryId);
+            });
+        };
+
+        scoringServiceObjApi.getSectionScoreResult = function (questionsResults, typeId, subjectId) {
+            return _getScoreTableProm().then(function (scoreTable) {
+                return _getSectionScoreResultFn(scoreTable, questionsResults, typeId, subjectId);
+            });
+        };
+
+        scoringServiceObjApi.rawScoreToScore = function (subjectId, rawScore) {
+            return _getScoreTableProm().then(function (scoreTable) {
+                var roundedRawScore = Math.round(rawScore);
+                return _getDataFromTable(scoreTable, keysMapConst.test, subjectId, roundedRawScore);
+            });
+        };
+
+        scoringServiceObjApi.getTotalScoreResult = function (scoresArr) {
+            var totalScores = 0;
+            angular.forEach(scoresArr, function (score) {
+                totalScores += score;
+            });
+            return $q.when(totalScores);
+        };
+
+        scoringServiceObjApi.getScoringSettings = function() {
```

Comment:
im not using the settings in the scoring service in any way, the only ones who use it, it's other components like diagnostic and userGoals for example